libmuscle/cpp/src/libmuscle/timestamp.cpp

```diff
@@ -10,20 +10,19 @@ using wallclock = std::chrono::high_resolution_clock;
 
 namespace libmuscle { namespace impl {
 
-Timestamp::Timestamp(double seconds)
-    : seconds(seconds)
-{
-    tzset();
-}
-
-Timestamp Timestamp::now() {
+Timestamp::Timestamp() {
     auto since_epoch = wallclock::now().time_since_epoch();
     double cycles = since_epoch.count();
-    double seconds = cycles * wallclock::period::num / wallclock::period::den;
-    return Timestamp(seconds);
+    seconds = cycles * wallclock::period::num / wallclock::period::den;
```

Comment:
Agreed, looks much better.

Funnily enough I ended up making a separate ProfilingTimestamp that uses integers to store the time for better precision, so this doesn't affect the profiling any more, but since the code was touched we might as well fix it while we're at it.